frida_tools/tracer.py

```diff
@@ -345,15 +345,35 @@ def _create_resolver_script(self):
     return workingSet;
 }
 
+function parseExportsFunctionPattern(pattern) {
+    var res = pattern.split('!');
+    var m, f;
+    
+    if (res.length === 1) {
+        m = '*';
+        f = res[0];
+    } else {
+        m = (res[0] === '') ? '*' : res[0];
+        f = (res[1] === '') ? '*' : res[1];
+    }
+    
+    return {
+        module: m,
+        function: f
+    };
+}
```

Comment:
This function should be moved below the function using it. (Code should flow from highest level to lowest level, so the reader can start at the top and read downwards, going from highest level things at the top all the way to the lowest level details at the bottom.)

Reply:
Gotcha.  I think I'm still influenced by C where a function needs to be defined before it is used! :-S

Reply:
Done.